### DISCUSSION TITLE
py2app launch error - can't open artisan app.

### DISCUSSION THREAD

**Discussion by a3noble** [artisan-roaster-scope/artisan#1154] [Q&A] [answered]
Hello! 
I'm super new to artisan and having issues opening the app. Trying to use version 2.8.2 on macOS 11.7.6.

When I open by clicking on the app icon I get this error message:

<img width="262" alt="launch error" src="https://user-images.githubusercontent.com/131686672/234036970-e1ad06e9-f097-4328-ad12-0d9af9501100.png">

Opening the app through the terminal gives me this:

```
Last login: Mon Apr 24 10:00:24 on ttys000
andrewnoble@MacBook-Pro ~ % /Volumes/Artisan/Artisan.app/Contents/MacOS/Artisan ; exit;
Traceback (most recent call last):
  File "/Volumes/Artisan/Artisan.app/Contents/Resources/__boot__.py", line 213, in <module>
    _run()
  File "/Volumes/Artisan/Artisan.app/Contents/Resources/__boot__.py", line 136, in _run
    exec(compile(source, path, "exec"), globals(), globals())
  File "/Volumes/Artisan/Artisan.app/Contents/Resources/artisan.py", line 76, in <module>
    from artisanlib import main, command_utility
  File "/Volumes/Artisan/Artisan.app/Contents/Resources/lib/python3.11/artisanlib/main.py", line 568, in <module>
    import appnope  # @UnresolvedImport # pylint: disable=import-error
    ^^^^^^^^^^^^^^
  File "appnope/__init__.pyc", line 13, in <module>
  File "appnope/_nope.pyc", line 13, in <module>
  File "ctypes/__init__.pyc", line 454, in LoadLibrary
  File "ctypes/__init__.pyc", line 376, in __init__
OSError: dlopen(/Users/andrewnoble/Library/Frameworks/Foundation.framework/Foundation, 6): no suitable image found.  Did find:
	/Users/andrewnoble/Library/Frameworks/Foundation.framework/Foundation: no matching architecture in universal wrapper
	/Users/andrewnoble/Library/Frameworks/Foundation.framework/Foundation: no matching architecture in universal wrapper
2023-04-24 10:04:17.071 Artisan[3594:84737] Launch error
2023-04-24 10:04:17.071 Artisan[3594:84737] Launch error
See the py2app website for debugging launch issues
```

I'm not really savvy enough to do much trouble shooting...due to the mention of python3.11, i checked what version of python i have and it's "Python 2.7.16". 

I assume this isn't an artisan app issue but is related to something on my computer. Pretty stuck - any help would be much appreciated!

**Comment by MAKOMO** [comment, marked as answer]
Hm. This is very strange. You are trying to start the official v2.8.2 build on your machine, right?
Does Artisan starts if opened from the Finder?

Those builds contain all libs required to run Artisan incl. Python 3.11. For what ever reason shared lib referenced from a lib named appnope used by Artisan tries to load some thing from 

/Users/andrewnoble/Library/Frameworks/Foundation.framework/Foundation

which seems to have  not the correct architecture for your Mac. Not sure why there is such a Foundation.framework at all installed under your user account. Maybe this was installed by some other software but with the wrong architecture.

You could try to create a new user account on your machine and see if Artisan runs on that one. You could also rename or delete this strange Foundation.framework folder but that might then break another software you run on your machine.

Sorry this is difficult to resolve on the Artisan side as the issue seems somehow caused by your macs unusual configuration.

**Reply by a3noble** [reply]
I created a new user account and was able to get Artisan to open without any issues! Really appreciate the suggestion!! 
To answer your questions, previously I was using the official v2.8.2 build and it didn't open from the Finder or the terminal window. Definitely some weird issue with my machine - thanks for helping me get around it!
